hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/compaction/log/TestCompactionFileInfo.java

```diff
@@ -0,0 +1,220 @@
+            "endRange",
+            "columnFamily",
+            "FileName is required parameter."
+        ),
+        Arguments.of("startRange is not present.",
+            "fileName",
+            null,
+            "endRange",
+            "columnFamily",
+            "Either all of startRange, endRange and columnFamily" +
+                " should be non-null or null. startRange: 'null', " +
+                "endRange: 'endRange', columnFamily: 'columnFamily'."
+        ),
+        Arguments.of("endRange is not present.",
+            "fileName",
+            "startRange",
+            null,
+            "columnFamily",
+            "Either all of startRange, endRange and columnFamily" +
+                " should be non-null or null. startRange: 'startRange', " +
+                "endRange: 'null', columnFamily: 'columnFamily'."
+        ),
+        Arguments.of("columnFamily is not present.",
+            "fileName",
+            "startRange",
+            "endRange",
+            null,
+            "Either all of startRange, endRange and columnFamily" +
+                " should be non-null or null. startRange: 'startRange', " +
+                "endRange: 'endRange', columnFamily: 'null'."
+        ),
+        Arguments.of("startRange and endRange are not present.",
+            "fileName",
+            null,
+            null,
+            "columnFamily",
+            "Either all of startRange, endRange and columnFamily" +
+                " should be non-null or null. startRange: 'null', " +
+                "endRange: 'null', columnFamily: 'columnFamily'."
+        ),
+        Arguments.of("endRange and columnFamily are not present.",
+            "fileName",
+            "startRange",
+            null,
+            null,
+            "Either all of startRange, endRange and columnFamily " +
+                "should be non-null or null. startRange: 'startRange', " +
+                "endRange: 'null', columnFamily: 'null'."
+        ),
+        Arguments.of("startRange and columnFamily are not present.",
+            "fileName",
+            null,
+            "endRange",
+            null,
+            "Either all of startRange, endRange and columnFamily" +
+                " should be non-null or null. startRange: 'null', " +
+                "endRange: 'endRange', columnFamily: 'null'."
+        )
+    );
+  }
+
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("compactionFileInfoInvalidScenarios")
+  public void testCompactionFileInfoInvalidScenario(String description,
+                                                    String fileName,
+                                                    String startRange,
+                                                    String endRange,
+                                                    String columnFamily,
+                                                    String expectedMessage) {
+    RuntimeException exception = Assertions.assertThrows(RuntimeException.class,
```

Comment:
```suggestion
    IllegalArgumentException exception = Assertions.assertThrows(IllegalArgumentException.class,
```

Reply:
It is `RuntimeException` on purpose. Because [this](https://github.com/apache/ozone/pull/5303/files#diff-f6807cfb3d601de6e4d002178c53f17767c9547ae04d19c16e089ececaea10a3R110) will throw conditional check NPE.